From 09ed2790db75c799a013646a1c74a4675157bf71 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Tue, 8 Mar 2016 05:17:06 -0500
Subject: [PATCH 0888/1110] drm/amd/dal: skip MST fake connectors on resume

This change is needed in order to avoid multiple dc_link_detect
calls on one DP link.

Also it disables removal of sink from connector hence allow
to restore mode on resume.

However this is only partial fix. Additional change will be
provided in order to fix topology change

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
index a9973ed..5bef63c 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
@@ -587,6 +587,14 @@ int amdgpu_dm_display_resume(struct amdgpu_device *adev )
 	list_for_each_entry(connector,
 			&ddev->mode_config.connector_list, head) {
 		aconnector = to_amdgpu_connector(connector);
+
+		/*
+		 * this is the case when traversing through already created
+		 * MST connectors, should be skipped
+		 */
+		if (aconnector->mst_port)
+			continue;
+
 		dc_link_detect(aconnector->dc_link, false);
 		aconnector->dc_sink = NULL;
 		amdgpu_dm_update_connector_after_detect(aconnector);
-- 
2.7.4

